cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -706,11 +706,11 @@ class EnumParser {
 };
 
 enum class TemporalComponent { kUnspecified = 0, kYear, kMonth, kDay, kSecond };
-static std::vector<std::string> kTemporalComponentOptions = {"YEAR", "MONTH", "DAY",
-                                                             "SECOND"};
+static std::vector<std::string> kTemporalComponentOptions = {"UNSPECIFIED", "YEAR",
+                                                             "MONTH", "DAY", "SECOND"};
```

Comment:
Why is there still a need for `UNSPECIFIED` here?

Reply:
Temporal extraction (in function tDecodeTemporalExtractionMapping) uses the kUnspecified state to determine if the enum argument was not passed to the function (to provide a more readable error message).  One alternative I considered was defaulting the value to kYear but that doesn't preserve the existing behavior.  I could remove kUnspecified if the less specific error is acceptable (I will have to investigate to see what the wording will be).

Reply:
Removing that parse extra parse check yields a rather useful "The value BOGUS is not an expected enum value" message when the value can't be parsed so I've removed it.